modules/nodes-containers-start-pod-safe-sysctls.adoc

```diff
@@ -0,0 +1,104 @@
+// Module included in the following assemblies:
+//
+// * nodes/containers/nodes-containers-sysctls.adoc
+
+:_content-type: PROCEDURE
+[id="nodes-starting-pod-safe-sysctls_{context}"]
+= Starting a pod with safe sysctls
+
+You can set sysctls on pods using the pod's `securityContext`. The `securityContext` applies to all containers in the same pod.
+
+Safe sysctls are allowed by default.
+
+This example uses the pod `securityContext` to set the following safe sysctls:
+
+* `kernel.shm_rmid_forced`
+* `net.ipv4.ip_local_port_range`
+* `net.ipv4.tcp_syncookies`
+* `net.ipv4.ping_group_range`
+
+[WARNING]
+====
+To avoid destabilizing your operating system, modify sysctl parameters only after you understand their effects.
+====
+
+Use this procedure to start a pod with the configured sysctl settings.
+[NOTE]
+====
+In most cases you will be modifying an existing pod definition and adding the `securityContext` spec.
+====
+
+
+.Procedure
+
+. Create a YAML file `sysctl_pod.yaml` that defines an example pod and add the `securityContext` spec, as shown in the following example:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sysctl-example
+  namespace: default
+spec:
+  containers:
+  - name: podexample
+    image: centos
+    command: ["bin/bash", "-c", "sleep INF"]
+  securityContext:
+    sysctls:
+    - name: kernel.shm_rmid_forced
+      value: "1"
+    - name: net.ipv4.ip_local_port_range
+      value: "32770       60666"
+    - name: net.ipv4.tcp_syncookies
+      value: "0"
+    - name: net.ipv4.ping_group_range
+      value: "0           200000000"
+----
++
+[NOTE]
+====
+An example pod is created here simply to illustrate how the changes are applied.
+====
+
+. Create the pod:
++
+[source,terminal]
+----
+$ oc apply -f sysctl_pod.yaml
+----
++
+. Verify that the pod is created:
++
+[source,terminal]
+----
+$ oc get pod
+----
++
+.Example output
+[source,terminal]
+----
+NAME              READY   STATUS            RESTARTS   AGE
+sysctl-example    1/1     Running           0          14s
+----
+
+. Log in to the pod:
++
+[source,terminal]
+----
+$ oc rsh sysctl-example
+----
+
+. Verify the values of the configured sysctl flags. For example find the value `kernel.shm_rmid_forced` as follows:
++
+[source,terminal]
+----
+sh-4.4# sysctl -a | grep kernel.shm_rmid_forced
```

Comment:
When I exec this command in unprivileged pod I receive the following output
```
> oc rsh sysctl-example
sh-4.4$ sysctl -a | grep kernel.shm_rmid_forced
sysctl: permission denied on key 'fs.protected_fifos'
sysctl: permission denied on key 'fs.protected_hardlinks'
sysctl: permission denied on key 'fs.protected_regular'
sysctl: permission denied on key 'fs.protected_symlinks'
sysctl: permission denied on key 'kernel.cad_pid'
sysctl: permission denied on key 'kernel.usermodehelper.bset'
sysctl: permission denied on key 'kernel.usermodehelper.inheritable'
kernel.shm_rmid_forced = 1
sysctl: permission denied on key 'net.ipv4.tcp_fastopen_key'
sysctl: permission denied on key 'net.ipv6.conf.all.stable_secret'
sysctl: permission denied on key 'net.ipv6.conf.default.stable_secret'
sysctl: permission denied on key 'net.ipv6.conf.eth0.stable_secret'
sysctl: permission denied on key 'net.ipv6.conf.lo.stable_secret'
sysctl: permission denied on key 'vm.mmap_rnd_bits'
sysctl: permission denied on key 'vm.mmap_rnd_compat_bits'
sysctl: permission denied on key 'vm.stat_refresh'
```
You can use the following command In order to avoid these errors:
`sysctl kernel.shm_rmid_forced`
output 
`kernel.shm_rmid_forced = 1`